src/csmake.pl

```diff
@@ -53,31 +54,43 @@
 }
 $instdir = abs_path($instdir);
 
-my $extract_targets = 0;
+# Copy arguments into MAKEARGS to use them with real make
+my @MAKEARGS = @ARGV;
 
-# Check CSMAKEFLAGS environment variable
+# Expand command-line arguments to include the CSMAKEFLAGS environment variable
 if ($ENV{'CSMAKEFLAGS'}) {
 	@ARGV = (split(/\s+/, $ENV{'CSMAKEFLAGS'}), @ARGV);
 }
 
 # Parse arguments
-if (@ARGV > 0 && $ARGV[0] eq "-t") {
-	$extract_targets = 1;
-	shift @ARGV;
+my %options=();  # csmake options
+{
+    local $SIG{__WARN__} = sub { };  # Supress warnings
```

Comment:
We shouldn't be needing this.

Reply:
If I remove it, it will print some warning messages that we don't need. For example, it will print that some of make's options are unknown (here we want to parse csmake's options) which may be a confusing.